app/views/handlers/list_action.py

```diff
@@ -1,39 +1,43 @@
 from flask import url_for
+from werkzeug.datastructures import ImmutableDict
 
 from app.questionnaire.location import Location
+from app.questionnaire.routing_path import RoutingPath
 from app.views.handlers.question import Question
 
 
 class ListAction(Question):
     @property
-    def parent_block(self):
+    def parent_block(self) -> ImmutableDict:
         parent_block_id = self._schema.parent_id_map[self.block["id"]]
-        return self._schema.get_block(parent_block_id)
+        # type ignore: get_block is being called with a valid block_id
+        return self._schema.get_block(parent_block_id)  # type: ignore
 
     @property
-    def parent_location(self):
+    def parent_location(self) -> Location:
         parent_block_id = self._schema.parent_id_map[self.block["id"]]
         return Location(
             section_id=self._current_location.section_id, block_id=parent_block_id
         )
 
-    def _get_routing_path(self):
+    def _get_routing_path(self) -> RoutingPath:
         return self.router.routing_path(section_id=self.parent_location.section_id)
 
-    def is_location_valid(self):
+    def is_location_valid(self) -> bool:
         can_access_parent_location = self.router.can_access_location(
             self.parent_location, self._routing_path
         )
 
         if (
```

Comment:
Can be simplified to something like and not sure you need the ignore here?
```suggestion
         return bool(
            can_access_parent_location
            and self._current_location.list_name == self.parent_block["for_list"]
        )
```

Reply:
Nice! Have changed this